src/screens/LocalConfirmNotificationCodeScreen.tsx

```diff
@@ -4,57 +4,68 @@ import React, { PureComponent } from 'react';
 import { View, Text, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
 
-import { CodeInput, Header, ScreenTemplate, Button, FlatButton } from 'app/components';
+import { CodeInput, Header, ScreenTemplate, Button, TimeoutButton } from 'app/components';
 import { Route, CONST, RootStackParams } from 'app/consts';
 import { ApplicationState } from 'app/state';
 import { selectors as notificationSelectors } from 'app/state/notifications';
+import {
+  verifyNotificationEmail as verifyNotificationEmailAction,
+  VerifyNotificationEmailActionFunction,
+} from 'app/state/notifications/actions';
 import { palette, typography } from 'app/styles';
 
 const i18n = require('../../loc');
 
 type State = {
   userCode: string;
   numberAttempt: number;
-  error: string;
+  localError: string;
 };
 
 interface Props {
   navigation: StackNavigationProp<RootStackParams, Route.LocalConfirmNotificationCode>;
   route: RouteProp<RootStackParams, Route.LocalConfirmNotificationCode>;
   pin: string;
+  error: string;
+  email: string;
+  verifyNotificationEmail: VerifyNotificationEmailActionFunction;
 }
 
 class LocalConfirmNotificationCodeScreen extends PureComponent<Props, State> {
   state = {
     userCode: '',
     numberAttempt: 0,
-    error: '',
+    localError: '',
   };
 
   setCode = (userCode: string) => {
     this.setState({ userCode });
   };
 
-  resendCode = () => {
-    this.setState({
-      error: '',
-      userCode: '',
-      numberAttempt: 0,
+  resendCode = (error = '') => {
+    const { verifyNotificationEmail } = this.props;
+    const { email } = this.props.route.params;
+
+    verifyNotificationEmail(email, {
```

Comment:
It's actually not about notification email, but just about verifying email. The endpoint is only to verify whether the user passed a valid email address that he has access to. So, I'd rename it but it's just a detail

Reply:
it depends how you read it, verify notification email means verify email for notification, we could have an email for a different feature for example verify airdrop email (of course we won't have, just an example)